### PR TITLE
Closes #2311 attend to `derive_vars_query()` bug

### LINKED ISSUE #2311: Bug: `derive_vars_query()` changes the AE terms to upper case
### What happened?

`derive_vars_query()` changes the AE terms to upper case, i.e., if the terms in AE and in the query dataset are mixed case, no terms are flagged.

There are two options to solve the issue:
- The case of the terms is not changed.
- The case of the terms is changed in both `dataset` and `dataset_queries`.

@pharmaverse/admiral , what do you think? Which option should be implemented?

### Session Information

_No response_

### Reproducible Example

```
library(admiral)

smq99 <- query(
  prefix = "SMQ06",
  id = auto,
  add_scope_num = TRUE,
  definition = basket_select(
    name = "Peripheral neuropathy (SMQ)",
    scope = "BROAD",
    type = "smq"
  )
)

# The get_terms function from admiralroche is used to get SMQ terms from OCEAN
aag_smq <- create_query_data(
  queries = list(smq99),
  get_terms_fun = admiralroche::get_terms,
  version = "26.1"
) %>%
  # version (VERSION) - add required prefix
  mutate(VERSION = paste0("MedDRA v", "26.1"))

test <- tibble::tribble(
  ~USUBJID, ~AEDECOD,
  "001", "Paraesthesia",
  "002", "Phrenic nerve paralysis",
)

test1 <- test %>% 
  derive_vars_query(
    aag_smq)
  )
```
does not flag any terms.
If
```
test1 <- test %>% 
  derive_vars_query(
    aag_smq %>% mutate(TERMCHAR = stringr::str_to_upper(TERMCHAR))
  )
```
is used, the terms are flagged correctly.

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by zdz2101** [comment]
@bms63 Is this the right way to do it? Create a branch based on the current release (v.1.0.0) , patch it, release/tag from it. Otherwise we can't directly commit and release from main right?

**Comment by bms63** [comment]
This feels good to me!  I like the idea of just using that `current_release` method.  We can then just get rid of the `patch` branch.

**Comment by bms63** [comment]
I don't we should patch directly to main and release.

**Comment by zdz2101** [comment]
I guess the problem here is this branch doesn't have the CI/CD stuff, I guess technically this is no different than using our `patch` branch right? But we need to catch patch up to `v1.0.0` ?

**Comment by bms63** [comment]
ah is patch still in the workflows?  maybe patch will never die

**Comment by bms63** [comment]
if not we can add `current_release` to get the worklows rolling

**Comment by zdz2101** [comment]
@bms63 made new `patch` branch based off of v1.0.0, this PR should contain the changes necessary for `v1.0.1` and then we can release the package from `patch` with `v1.0.1` release tag? 


@bundfussr Do you have an appropriate reprex that does not depend on `admiralroche` ?

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# admiral 1.0.1
+
+- Fix bug in `derive_vars_query()`. (#2311)
```
Comment (bms63) [inline NEWS.md:3]: Can we be a little more specific in the big we fixed please

**Comment by bms63** [comment]
@zdz2101 when you have a second could you fix the links please

@dgrassellyb the check templates action is failing.  any ideas?

@millerg23, @rossfarrugia, @bundfussr any final words, blessings, sacrifices to the CRAN gods?

**Review by bundfussr — COMMENTED** [review]
I think we should mention in the documentation of `derive_vars_query()` that the matching is performed case insensitive?

**Comment by bms63** [comment]
> I think we should mention in the documentation of `derive_vars_query()` that the matching is performed case insensitive?

Ohh -- good call!!!

**Review by bms63 — APPROVED** [review]
Thanks @zdz2101

**Comment by bms63** [comment]
@zdz2101 do we need the skip vbump here?

**Comment by bms63** [comment]
> @zdz2101 do we need the skip vbump here?

looks like vbump only acts on main
image

**Comment by bms63** [comment]
Steps loosely:

- [ ] CRAN accepts we release from `patch`.  
- [ ] Release triggers a rebuild of the website for `1.0.1`.  
- [ ] Check website has updated correctly
- [ ] Merge patch into main (careful with renv, renv.lock, etc as this is still in `patch - just want to make sure this is removed) 
- [ ] Continue on with dev work

**Comment by zdz2101** [comment]
> Steps loosely:
> 
> * [ ]  CRAN accepts we release from `patch`.
> * [ ]  Release triggers a rebuild of the website for `1.0.1`.
> * [ ]  Check website has updated correctly
> * [ ]  Merge patch into main (careful with renv, renv.lock, etc as this is still in `patch - just want to make sure this is removed)
> * [ ]  Continue on with dev work

A+ I understand

**Pull request merged**